pom.xml

```diff
@@ -42,10 +42,22 @@
     <maven.compiler.release>11</maven.compiler.release>
     <junit-jupiter.version>5.5.2</junit-jupiter.version>
     <opencensus.version>0.24.0</opencensus.version>
+    <auto.value.version>1.2</auto.value.version>
+    <com.google.guava>28.0-jre</com.google.guava>
   </properties>
 
   <dependencyManagement>
     <dependencies>
+      <dependency>
+        <groupId>com.google.guava</groupId>
+        <artifactId>guava</artifactId>
+        <version>${com.google.guava}</version>
+      </dependency>
+      <dependency>
+        <groupId>com.google.auto.value</groupId>
```

Comment:
Can we avoid using auto value and try keep this client lean?

Reply:
I think there is a value in using auto value we don't have to write those crucial boilerplate methods and the code is more readable.